catowseriOS/CottonPlugins/JavaScriptEvaluateble.swift

```diff
@@ -28,27 +51,32 @@ extension JavaScriptEvaluateble {
     }
 
     @available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
-    func evaluatePublisher(jsScript: String) -> AnyPublisher<Any, Error> {
-        let p = Future<Any, Error> { [weak self] (promise) in
+    func evaluatePublisher(jsScript: String) -> AnyPublisher<String, Error> {
+        let future = Future<String, Error> { @MainActor @Sendable [weak self] promise in
             guard let self = self else {
                 promise(.failure(CottonPluginError.zombiError))
                 return
             }
-            self.evaluateJavaScript(jsScript) { (something, error) in
-                if let realError = error {
-                    promise(.failure(realError))
+#if swift(>=6)
+            // workaround until Apple Combine fixes the future/promise by adding Sendable conformance
+            nonisolated(unsafe) let promise = promise
```

Comment:
Use #warning macro instead, also check if it compiles on swift 5.7 or any swift < 6.0